db/db_local/start_database.go

```diff
@@ -134,18 +140,85 @@ func StartDB(port int, listen StartListenType, invoker constants.Invoker) (start
 	return ServiceStarted, err
 }
 
-// startPostgresProcess starts the postgres process and writes out the state file
+// startPostgresProcessAndSetPassword starts the postgres process and writes out the state file
 // after it is convinced that the process is started and is accepting connections
-func startPostgresProcess(port int, listen StartListenType, invoker constants.Invoker) error {
+func startPostgresProcessAndSetPassword(port int, listen StartListenType, invoker constants.Invoker) (e error) {
 	utils.LogTime("startPostgresProcess start")
 	defer utils.LogTime("startPostgresProcess end")
 
+	defer func() {
+		if e != nil {
+			// remove the state file if we are going back with an error
+			removeRunningInstanceInfo()
+		}
+	}()
+
 	listenAddresses := "localhost"
 
 	if listen == ListenTypeNetwork {
 		listenAddresses = "*"
 	}
 
+	if err := writePGConf(); err != nil {
+		return err
+	}
+
+	postgresCmd := createCmd(port, listenAddresses)
+	setupLogCollection(postgresCmd)
+	err := postgresCmd.Start()
+	if err != nil {
+		return err
+	}
+
+	// get the password from the password file
+	password, err := readPasswordFile()
+	if err != nil {
+		return err
+	}
+
+	// if a password was set through the `STEAMPIPE_DATABASE_PASSWORD` environment variable
+	// or through the `--database-password` cmdline flag, then use that for this session
+	// instead of the default one
+	if viper.IsSet(constants.ArgServicePassword) {
+		password = viper.GetString(constants.ArgServicePassword)
+	}
+
+	err = createRunningInfo(postgresCmd, port, password, listen, invoker)
+	if err != nil {
+		postgresCmd.Process.Kill()
+		return err
+	}
+
+	err = postgresCmd.Process.Release()
+	if err != nil {
+		postgresCmd.Process.Kill()
+		return err
+	}
+
+	err = setupServicePassword(invoker, password)
+	if err != nil {
+		postgresCmd.Process.Kill()
+		return err
+	}
+	return nil
+}
+
+func setupServicePassword(invoker constants.Invoker, password string) error {
```

Comment:
can you ensure these functions appear in the order they are called in startPostgresProcessAndSetPassword